syntax = "proto3";

import "pipelines.proto";
import "util.proto";
import "pm_models.proto";

package ficus;

service BackendService {
    rpc ExecutePipeline(PipelineExecutionRequest) returns (GrpcGuid);

    rpc GetContextValue(GetContextValueRequest) returns (ContextValue);
}

message GetContextValueRequest {
    GrpcGuid executionId = 2;
    ContextKey key = 1;
}

message PipelineExecutionRequest {
    Pipeline pipeline = 1;
}

